src/solidlsp/language_servers/sourcekit_lsp.py

```diff
@@ -0,0 +1,561 @@
+import logging
+import os
+import pathlib
+import subprocess
+import threading
+
+from overrides import override
+
+from solidlsp.ls import SolidLanguageServer
+from solidlsp.ls_config import LanguageServerConfig
+from solidlsp.ls_logger import LanguageServerLogger
+from solidlsp.lsp_protocol_handler import lsp_types
+from solidlsp.lsp_protocol_handler.lsp_types import InitializeParams
+from solidlsp.lsp_protocol_handler.server import ProcessLaunchInfo
+from solidlsp.settings import SolidLSPSettings
+from solidlsp.lsp_protocol_handler import lsp_types as LSPTypes
+
+
+class SourceKitLSP(SolidLanguageServer):
+    """
+    Provides Swift specific instantiation of the LanguageServer class using sourcekit-lsp.
+    """
+
+    @override
+    def is_ignored_dirname(self, dirname: str) -> bool:
+        # For Swift projects, we should ignore:
+        # - .build: Swift Package Manager build artifacts
+        # - .swiftpm: Swift Package Manager metadata
+        # - node_modules: if the project has JavaScript components
+        # - dist/build: common output directories
+        return super().is_ignored_dirname(dirname) or dirname in [".build", ".swiftpm", "node_modules", "dist", "build"]
+
+    @staticmethod
+    def _get_swift_version():
+        """Get the installed Swift version or None if not found."""
+        try:
+            result = subprocess.run(["swift", "--version"], capture_output=True, text=True, check=False)
+            if result.returncode == 0:
+                return result.stdout.strip()
+        except FileNotFoundError:
+            return None
+        return None
+
+    @staticmethod
+    def _get_sourcekit_lsp_version():
+        """Get the installed sourcekit-lsp version or None if not found."""
+        try:
+            result = subprocess.run(["sourcekit-lsp", "-h"], capture_output=True, text=True, check=False)
+            if result.returncode == 0:
+                return result.stdout.strip()
+        except FileNotFoundError:
+            return None
+        return None
+
+    @staticmethod
+    def _setup_runtime_dependency():
+        """
+        Check if required Swift runtime dependencies are available.
+        Raises RuntimeError with helpful message if dependencies are missing.
+        """
+        swift_version = SourceKitLSP._get_swift_version()
+        if not swift_version:
+            raise RuntimeError(
+                "Swift is not installed. Please install Swift from https://swift.org/download/ and make sure it is added to your PATH."
+            )
+
+        sourcekit_lsp_version = SourceKitLSP._get_sourcekit_lsp_version()
```

Comment:
Most of our LS download their dependencies by themselves, would that not be possible for sourcekit?